pkg/deviceplugin/lister.go

```diff
@@ -0,0 +1,74 @@
+package deviceplugin
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/kubevirt/device-plugin-manager/pkg/dpm"
+)
+
+const (
+	resourceNamespace         = "macvtap.network.kubevirt.io"
+	ConfigEnvironmentVariable = "DP_MACVTAP_CONF"
+)
+
+type MacvtapConfig struct {
+	Name     string `json:"name"`
+	Master   string `json:"master"`
+	Mode     string `json:"mode"`
+	Capacity int    `json:"capacity"`
+}
+
+type MacvtapLister struct {
+}
+
+func (ml MacvtapLister) GetResourceNamespace() string {
+	return resourceNamespace
+}
+
+func readConfig() (map[string]MacvtapConfig, error) {
+	var config []MacvtapConfig
+	configMap := make(map[string]MacvtapConfig)
+
+	configEnv := os.Getenv(ConfigEnvironmentVariable)
+	err := json.Unmarshal([]byte(configEnv), &config)
+	if err != nil {
+		return configMap, err
+	}
+
+	for _, macvtapConfig := range config {
+		configMap[macvtapConfig.Name] = macvtapConfig
+	}
+
+	return configMap, nil
+}
+
+func (ml MacvtapLister) Discover(pluginListCh chan dpm.PluginNameList) {
+	var plugins = make(dpm.PluginNameList, 0)
```

Comment:
So to apply new config, all DaemonSets need to be restarted? Could you document it?

Reply:
Yes, ideally deleting the config map and creating a new one would take care of it. I was waiting for the cni to wire an encompassing readme to include this information.